dolphin/drivers/manager.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from oslo_log import log
+
+
+LOG = log.getLogger(__name__)
+
+DRIVER_MAPPING = {
+    "fake_storage": "dolphin.drivers.fake_storage.FakeStorageDriver"
+}
+
+
+class DriverManager(object):
+
+    def __init__(self):
+        # The driver_factory will keep the driver instance for
+        # each of storage systems so that the session between driver
+        # and storage system is effectively used.
+        self.driver_factory = dict()
+
+    @staticmethod
+    def get_storage_registry():
+        """Show register parameters which the driver needs."""
+        pass
+
+    def register_storage(self, context, register_info):
+        """Discovery a storage system with register parameters."""
+        pass
+
+    def remove_storage(self, context, storage_id):
+        """Clear driver instance from driver factory."""
+        self.driver_factory.pop(storage_id, None)
+
+    def get_storage(self, context, storage_id):
+        """Get storage device information from storage system"""
+        pass
+
+    def list_pools(self, context, storage_id):
```

Comment:
does context conatins register info ? if so  do we need storage_id separate?

Reply:
Context contains the common request parameter, may be used in future. it doesn't contains register info, it's not good to save/pass register info in memory except the register api.